lib/game.ex

```diff
@@ -124,4 +128,128 @@ defmodule Game do
   def get_marker(game), do: game.player_2_mark
 
   def update_board(board, game), do: %{game | board: board}
+
+  def next_open(game), do: List.first(get_available_moves(game))
+
+  def handle_next_move(game) do
+    move = next_open(game)
+    player_move_list = determine_move_list(game)
+    game = update_available_moves(move, game)
+    game = update_player_move_list(game, player_move_list, move)
+    place_marker(move, game)
+  end
+
+  def best_move(game) do
+    # best_score = -100
+    # new_move = nil
+    game = assign_maximizer(game)
+
+    # Iterate through every cell
+    score_move_list =
+      Enum.map([1, 2, 3, 4, 5, 6, 7, 8, 9], fn move ->
+        IO.puts("\r\n\r\nTrying == #{move} ")
+
+        # Grab that move number in case we want to use it
+        IO.puts("AVAIL MOVES BELOW")
+        IO.inspect(game.available_moves)
+        IO.puts("AVAIL MOVES ABOVE")
+
+        IO.puts("MOVE IN AVAIL MOVES?")
+        IO.inspect(move in game.available_moves)
+        IO.puts("MOVE IN AVAIL MOVES?")
+
+        # If move is in available moves continue
+        if move in game.available_moves do
+          # IO.puts("BEST SCORE BEFORE ==== #{best_score}")
+          IO.puts("DOING == #{move} ")
+          # play the next available move
+          game = handle_move(move, game)
+          game = check_for_win(game)
+          {move, score(game)}
+
+        else
+          # If move is not in available moves skip
+          IO.puts("SKIPPING == #{move}\n")
+        end
+      end)
+
+    move_list = Enum.filter(score_move_list, fn elem -> elem != :ok end)
+    list_of_scores = Enum.map(move_list, fn {_move, score} -> score end)
+    max_score = Enum.max(list_of_scores)
+    index_of_max_score = Enum.find_index(move_list, fn {_move, score} -> score == max_score end)
```

Comment:
this feels very imperative to me 😄

Reply:
It is insanely ugly and imperative. I hate it, but what recommendations do you have to try to make it not imperative? Any guidance on things to try to get me out of the imperative thinking here? 

Not really looking for an answer, but suggestions on ways of thinking would be great! Trying to break old habits of OOP.

Reply:
so making something pipeable is usually a good start, and losing the trail of variable assignments. that way you're composing functions together, which kind of forces you to write declaratively out of the box.

a simplified of what you're trying to do here would be:

```elixir
defmodule FP do
  def get_best_move(moves, scores) do
    {_, move} = Enum.zip(scores, moves)
    |> Enum.max()

    move
  end
end

moves = [1,2,3,4,5]
scores = [0,0,10,0,-10]

FP.get_best_move(moves, scores)
|> IO.inspect() # 3
``` 

notice we're relying on functions to get everything we need, as opposed to imperatively building everything into a variable, finding the index, then combining everything together.